universe/ValueRefs.cpp

```diff
@@ -248,7 +249,47 @@ namespace {
     const std::string EMPTY_STRING;
 }
 
+// helper: support enums in ValueRef<T>::StringResult() via ADL
+// c++11 direct use of std::enable_if would confuse the signature of StringResult
+// c++17 could use: if constexpr (std::is_enum<T>::value
+template <typename T>
```

Comment:
should this be in an anonymous namespace?

Reply:
Seems ADL does not work if i put it in anonymous namespace